### ISSUE TITLE
Failling back to canonical URL?

### ISSUE THREAD

**Issue by julien51** [jhass/open_graph_reader#5]
I know url is a required OG element, but as you said in the comments yourself, it's often missing from documents. Any chance we could fallback on the canonical  (or self) URL when that's the case?

**Comment by jhass** [comment]
Yes, just set [synthesize_url](http://www.rubydoc.info/gems/open_graph_reader/OpenGraphReader/Configuration#synthesize_url-instance_method) to `true`.

**Comment by julien51** [comment]
Hum, this is what I did... it does use the request's URL but that's not the canonical URL from `<link rel="canonical"  >`...

**Comment by jhass** [comment]
Hm, that's a widespread thing? I never noticed it. I would accept a patch that prefers it over the request URL if present, you would need to expose it in the `Parser` and then handle it [here](https://github.com/jhass/open_graph_reader/blob/master/lib/open_graph_reader/builder.rb#L127).
